x/stakeibc/keeper/msg_server_instant_redeem_stake.go

```diff
@@ -0,0 +1,110 @@
+package keeper
+
+import (
+	"context"
+	sdkmath "cosmossdk.io/math"
+	"fmt"
+	"github.com/Stride-Labs/stride/v5/utils"
+	recordstypes "github.com/Stride-Labs/stride/v5/x/records/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v5/x/stakeibc/types"
+)
+
+func (k msgServer) InstantRedeemStake(goCtx context.Context, msg *types.MsgInstantRedeemStake) (*types.MsgInstantRedeemStakeResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	k.Logger(ctx).Info(fmt.Sprintf("fast unbond: %s", msg.String()))
+
+	// get our addresses, make sure they're valid
+	sender, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "creator address is invalid: %s. err: %s", msg.Creator, err.Error())
+	}
+
+	// then make sure host zone is valid
+	hostZone, found := k.GetHostZone(ctx, msg.HostZone)
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidHostZone, "host zone is invalid: %s", msg.HostZone)
+	}
+
+	// safety check: redemption rate must be within safety bounds
+	rateIsSafe, err := k.IsRedemptionRateWithinSafetyBounds(ctx, hostZone)
+	if !rateIsSafe || (err != nil) {
+		errMsg := fmt.Sprintf("IsRedemptionRateWithinSafetyBounds check failed. hostZone: %s, err: %s", hostZone.String(), err.Error())
+		return nil, sdkerrors.Wrapf(types.ErrRedemptionRateOutsideSafetyBounds, errMsg)
+	}
+
+	// construct desired unstaking amount from host zone
```

Comment:
I think we should add some logic here to charge a fee on `InstantRedeemStake` (even though initial fee will be zero), just to make it easier to publicly signal that this won't be a free feature forever.

I think this should just be defining some `feeAmount` here, then creating a new `RedeemAmount` that is `msg.Amount - feeAmount`, and then transferring the fee to the community pool at the end of the function.

Reply:
So an absolute amount, not a percentage then?  Also, are you thinking this is just a constant in code to signal intention, or a module parameter?

Reply:
Oh apologies, `feeAmount` above would be a calculated value, I think the fee should be a percentage. 

I think we should make it a module parameter, to make it a bit more explicit (and this _feels_ like a variable that governance should be allowed to change). 

Let me know what you think though!

Reply:
This has been implemented, although not tested specifically for non-zero values.  Let me know if you still have any issues here.